### PR TITLE
Implement MkTable#delete()

### LINKED ISSUE #35: MkTable.java:106-107: Implement MkTable.delete() operation. delete() method...
Puzzle `8-abdd4627` in `src/main/java/com/jcabi/dynamo/mock/MkTable.java:106-107` has to be resolved: Implement MkTable.delete() operation. delete() method added to Table interface and implemented in AwsTable class so far.... The puzzle was created by Nikolay Popov on 28-Jan-2015.

If you have any technical questions, don't ask me, submit new tickets instead

### PR DESCRIPTION
**Pull request by Suseika**
I had to implement method `MkTable#delete()`. 

In order to do so, I:
- created a new method `MkData#delete`;
- added a puzzle in `H2Data` to implement `H2Data#delete()` because it implements `MkData`.

Fixes #35

### CODE REVIEW

**Comment by Suseika** [comment]
@dmarkov Please find a reviewer.

**Comment by dmarkov** [comment]
@suseika Let me find a reviewer for this pull request, thanks for submitting it

**Comment by dmarkov** [comment]
@pinaf this pull request is for you, please review

**Comment by Suseika** [comment]
@pinaf Are you going to review this PR? If not, please ask to assign another reviewer to this PR.

File: src/main/java/com/jcabi/dynamo/mock/MkTable.java [hunk 1]
```diff
@@ -111,6 +111,9 @@ public String name() {
     @Override
```
Comment (pinaf) [inline src/main/java/com/jcabi/dynamo/mock/MkTable.java]: @Suseika remember to remove the puzzle above

File: src/main/java/com/jcabi/dynamo/mock/MkData.java [hunk 2]
```diff
@@ -73,4 +73,11 @@
     void update(String table, Attributes keys,
         AttributeUpdates attrs) throws IOException;
 
+    /**
+     * Delete attributes from the given table.
+     * @param table Table name
+     * @param keys Keys
+     * @throws IOException If fails
+     */
+    void delete(String table, Attributes keys) throws IOException;
```
Comment (pinaf) [inline src/main/java/com/jcabi/dynamo/mock/MkData.java]: @Suseika let's rename `keys` to `attrs`

File: src/main/java/com/jcabi/dynamo/mock/H2Data.java [hunk 3]
```diff
@@ -259,6 +259,20 @@ public void update(final String table, final Attributes keys,
     }
 
     /**
+     * Delete attributes from the given table.
+     * @todo 30min:#35 Implement H2Data#delete() method because MkData#delete
```
Comment (pinaf) [inline src/main/java/com/jcabi/dynamo/mock/H2Data.java]: @Suseika the puzzle syntax is wrong here. first the issue number and then the time `@todo #35:30min ...`

**Comment by pinaf** [comment]
@Suseika 2 comments above.

**Comment by pinaf** [comment]
@Suseika I don't think much was accomplished with this PR. Can't you implement the `H2Data#delete` method and add a test?

**Comment by Suseika** [comment]
@pinaf 
> I don't think much was accomplished with this PR. Can't you implement the H2Data#delete method and add a test?

Sorry, but I don't think that would be a valid reason to reject a PR as it is. My task specifically was to implement `MkTable#delete()`, and I implemented that. Implementing `H2Data#delete()` would be another task. I think I did exactly what I was asked to do.

**Comment by Suseika** [comment]
@pinaf One new commit above.

**Comment by pinaf** [comment]
@rultor merge

**Comment by pinaf** [comment]
@yegor256 I don't agree with this PR as I stated above to @Suseika. Please decide as you are this project's architect.

**Comment by Suseika** [comment]
@pinaf Ok, I implemented `H2Data#delete` as you asked. But Appveyor's build in failing for some reason, and Travis build doesn't even start. It builds on my machine though, should build on rultor.

**Comment by pinaf** [comment]
@Suseika I'll take a look but it seems like there are merge conflicts

**Comment by Suseika** [comment]
@pinaf You were right, there were merge conflicts. Appveyor reported them correctly, but it just seemed suspicious to me that Travis build didn't even start. I fixed the conflicts, please see the new commit.

**Comment by pinaf** [comment]
@rultor merge

**Comment by pinaf** [comment]
@Suseika thanks!

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@pinaf **22 mins** added to your account (payment number `AP-5KG647699B708294U`), many thanks for your contribution! 384 hours and 21 mins spent here.. extra minutes for review comments (c=7). +22 added to your rating, at the moment it is: [+5518](http://www.netbout.com/b/35128?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy pls

**Pull request merged**